code/game/objects/items.dm

```diff
@@ -127,6 +127,10 @@
 	if(ismob(loc))
 		dropped(loc)
 
+	for(var/obj/item/item in contents_recursive()) // Get rid of any intel objects so we don't delete them
+		if(item.is_objective)
```

Comment:
Do not check `is_objective` here. It is entirely separate variable (and I have some unrelated plans that might affect it in the future).

In fact, instead of removing bits of code in one place and adding them in the other, it is probably simpler to undo this addition entirely...